### PR TITLE
ci: add debianize build pipeline

### PR DESCRIPTION
**Pull request by yedpodtrzitko**
### add Debian package build into pipeline

This was quite a bamboozle as the Debian building somehow feels like it does not behave deterministically - one day it is working certain way and when I pick it up the next day where I left it, it behaves differently. 
Specifically - one day it was pinning the exact version of Debian dependencies. The next day it was not pinning them, and I couldnt reproduce the previous behaviour again. This made it easier for installation, but also it requires to:
- add manually pinned `b2sdk` dependency into `debian/control` as the latest available version of `b2sdk` Debian package is incompatible: (`ImportError: cannot import name 'UploadMode' from 'b2sdk.v2'` [which was added later](https://github.com/Backblaze/b2-sdk-python/blob/1d2001ca2e4868feff23fbfe8867f0ef5d99bc09/b2sdk/_v3/__init__.py#L188) 
- which also means we need an up-to-date Debian package of b2sdk, see reef-technologies/b2-sdk-python/pull/287

tl;dr - package build is working and it's uploaded into B2 bucket:

- https://f005.backblazeb2.com/file/reef-b2-debian/python3-b2sdk_1.21.0-1_all.deb

- https://f005.backblazeb2.com/file/reef-b2-debian/backblaze-b2_3.9.0-1_all.deb

### Repo Setup

- add the following pair of secrets for uploading the package into B2 bucket:
  - `B2_DEBIAN_APPLICATION_KEY_ID`
  - `B2_DEBIAN_APPLICATION_KEY`

- add the following environment variable to determine which B2 bucket upload will go to:
  - `B2_DEBIAN_BUCKET_UPLOAD`

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -158,3 +158,10 @@ jobs:
           context: .
           push: true
           tags: backblaze/b2:${{ steps.build.outputs.version }}
+
+  debian-release:
+    name: Build Debian Package
```
Comment (mpnowacki-reef) [inline .github/workflows/cd.yml:163]: let's skip this step if vars are missing

File: .github/workflows/cd_debian.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      b2-upload-bucket:
+        required: true
+        type: string
+
+env:
+  PYTHON_DEFAULT_VERSION: "3.11"
```
Comment (mpnowacki-reef) [inline .github/workflows/cd_debian.yml:11]: debian is probably decades away from supporting this python, I assume that's not a problem? like maybe some things here pass even though they are syntax errors in the version that debian is gonna use?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,13 +6,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+* Unpin `setuptools-scm<6.0` as we don't support Python 3.5 (reason for pinning) anymore
+* use `rst2ansi` conditionally as it's not available in Debian packaging
 
 ### Infrastructure
 * Autocomplete integration tests will now work properly even if tested package has not been installed
 * Automatically set copyright date when generating the docs
 * Increase timeout time in autocomplete tests to accommodate slower CI environments
 * Update pyinstaller to fix Linux Bundle build
 * Replace `pyflakes` with `ruff` for linting
+* Add Debian package build to CI
```
Comment (mpnowacki-reef) [inline CHANGELOG.md:19]:
```suggestion
* Add Debian package build to CD
```

File: b2/arg_parser.py [hunk 4]
```diff
@@ -13,10 +13,23 @@
...
 from b2sdk.v2 import RetentionPeriod
-from rst2ansi import rst2ansi
+
+rst2ansi = None
+with suppress(ImportError):
+    from rst2ansi import rst2ansi
+
+if not rst2ansi:
+
+    def rst2ansi(input_string, output_encoding='utf-8'):
```
Comment (mpnowacki-reef) [inline b2/arg_parser.py:27]: soooooooo, help messages in debian are gonna look bad? Wouldn't it be possible to dump the rst2ansi output when building debian packages and included that? Or is that a lot of work?

File: .github/workflows/cd_debian.yml (hunk 2 above)
Comment (yedpodtrzitko) [reply .github/workflows/cd_debian.yml:11]: 3.11 is actually the default version in the latest Debian.

But we run testsuite for all the supported Python versions (3.7+), so if anything would be misbehaving we'd find out there, no?

File: b2/arg_parser.py (hunk 4 above)
Comment (yedpodtrzitko) [reply b2/arg_parser.py:27]: I wanted to discuss the potential approach here. The option I'd be leaning towards the most (due to its trivial complexity) would be to strip here the "special" characters from the source and use the output plaintext. It was plaintext even before adding rst2ansi (?), so it's not like something is taken away. Assuming Debian will eventually catch up and add package for `rst2ansi` without us doing some complex workarounds for nice2have features.

> when building debian packages and included that? Or is that a lot of work?

This PR looks quite simple in the end, but doing _anything_ within the Debian packaging is so immeasurable PITA that I really dont want to do that.

**Review by mpnowacki-reef — APPROVED** [review]
